src/pages/projectSettings/tabs/ViewsAndFiltersTab/transformers.ts

```diff
@@ -1,40 +1,45 @@
 import { ProjectSettingsTabRoutes } from "constants/routes";
-import { ProjectSettingsQuery } from "gql/generated/types";
 import { FormToGqlFunction, GqlToFormFunction } from "../types";
+import { ProjectType } from "../utils";
 
 type Tab = ProjectSettingsTabRoutes.ViewsAndFilters;
 
-export const gqlToForm = ((data: ProjectSettingsQuery["projectSettings"]) => {
+export const gqlToForm = ((data, { projectType }) => {
   if (!data) return null;
 
-  const {
-    projectRef: { parsleyFilters, projectHealthView },
-  } = data;
+  const { projectRef } = data;
 
   return {
     parsleyFilters:
-      parsleyFilters?.map(({ caseSensitive, exactMatch, expression }) => ({
-        displayTitle: expression,
-        expression,
-        caseSensitive,
-        exactMatch,
-      })) ?? [],
-    view: {
-      projectHealthView,
-    },
+      projectRef.parsleyFilters?.map(
+        ({ caseSensitive, exactMatch, expression }) => ({
+          displayTitle: expression,
+          expression,
+          caseSensitive,
+          exactMatch,
+        })
+      ) ?? [],
+    ...(projectType !== ProjectType.Repo &&
+      "projectHealthView" in projectRef && {
+        view: {
+          projectHealthView: projectRef.projectHealthView,
+        },
+      }),
   };
 }) satisfies GqlToFormFunction<Tab>;
 
-export const formToGql = (({ parsleyFilters, view }, id) => ({
+export const formToGql = ((formState, id) => ({
   projectRef: {
     id,
-    parsleyFilters: parsleyFilters.map(
+    parsleyFilters: formState.parsleyFilters.map(
       ({ caseSensitive, exactMatch, expression }) => ({
         expression,
         caseSensitive,
         exactMatch,
       })
     ),
-    projectHealthView: view.projectHealthView,
+    ...("view" in formState && {
+      projectHealthView: formState.view.projectHealthView,
+    }),
```

Comment:
Is it possible to keep the destructuring on L31 and then do:
```suggestion
    ...(view && {
      projectHealthView: view.projectHealthView,
    }),
```